### ISSUE TITLE
bug: IonButton aria-describedby doesn't work

### ISSUE THREAD

**Issue by kszepieto** [ionic-team/ionic-framework#28287]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#creating-an-issue).
- [X] I agree to follow the [Code of Conduct](https://ionicframework.com/code-of-conduct).
- [X] I have searched for [existing issues](https://github.com/ionic-team/ionic-framework/issues) that already report this problem, without success.

### Ionic Framework Version

v7.x

### Current Behavior

When using NVDA IonButton doesn't handle ```aria-describedby``` attribute correctly, I've added this code to blank ionic app

```
const ExploreContainer: React.FC = () => {
  return (
    <div id="container">
      <div>
        <input type="text"></input>
      </div>
      <h2 id="h2Id1"> Some content 1</h2>
      <h2 id="h2Id2">Some content 2</h2>

      <div>
        <IonButton aria-describedby="h2Id1">
          <span aria-describedby="h2Id1">IonButton 1</span>
        </IonButton>
        <IonButton aria-describedby="h2Id2">IonButton 2</IonButton>
      </div>
      
      <div>
        <button role="button" aria-describedby="h2Id1">
          Normal 1
        </button>
        <button role="button" aria-describedby="h2Id2">
          Normal 2
        </button>
      </div>
      <span tabIndex={0} role="button" aria-describedby="h2Id2">
        Span 1
      </span>
    </div>
  );
};
```

Tabbing on this app gives following results
![image](https://github.com/ionic-team/ionic-framework/assets/8173606/77bbeca5-df93-459a-8204-dd5692ec6824)

### Expected Behavior

When tabbing on IonButtons, they should handle ```aria-describedby``` property (1), Similarly to html buttons and span (2)

Tabbing on "IONBUTTON 1", NVDA should generate: _clickable IONBUTTON 1 button **Some content 1**_
Tabbing on "IONBUTTON 2", NVDA should generate: _clickable IONBUTTON 1 button **Some content 2**_

### Steps to Reproduce

Check Current behaviour section

### Code Reproduction URL

https://github.com/kszepieto/ButtonAria

### Ionic Info

Ionic:

   Ionic CLI       : 6.20.1 (C:\Users\krzysztof\AppData\Roaming\npm\node_modules\@ionic\cli)        
   Ionic Framework : @ionic/react 7.4.3

Capacitor:

   Capacitor CLI      : 5.4.2
   @capacitor/android : not installed
   @capacitor/core    : 5.4.2
   @capacitor/ios     : not installed

Utility:

   cordova-res : not installed globally
   native-run  : 1.7.3

System:

   NodeJS : v16.17.0 (C:\Program Files\nodejs\node.exe)
   npm    : 8.15.0
   OS     : Windows 10

**Comment by liamdebeasi** [comment]
Hey there,

While I understand this is not a desired behavior, `ion-button` is working as intended here. `ion-button` uses the Shadow DOM, and attributes such as `aria-describedby` do not work across shadow DOM boundaries as it would break encapsulation. 

As an example:

When you set `aria-describedby` that attribute needs to be set on the native `button` inside of the shadow root (since the `<button>` is what receives focus). However, since `aria-describedby` does not work across shadow dom boundaries, the reference element is never found.

```html
<p id="desc">Lorem Ipsum</p>

<ion-button>
  #shadow-root
    <button aria-describedby="desc">...</button>
</ion-button>
```

-----

You can use `aria-label` to provide more context for your `ion-button`. Alternatively, you can use a regular `<button>` element if you need to use `aria-describedby`. There are improvements to the Shadow DOM API that would let us reference describing elements across shadow root boundaries, though that work is still being implemented in browsers. I am going to close this, but let me know if you have any questions.
